### ISSUE TITLE
Sign In With Apple has a 'null' app name & NO app icon

### ISSUE THREAD

**Issue by ameedsayeh** [aws-amplify/amplify-ios#1095]
**Describe the bug**
Sign in with Apple Sheet presents app name as **"null"** and does not present the app icon.
The Login flow works perfectly, but the look does NOT.

**To Reproduce**
1. Xcode project configured with amplify and Cognito plugin. [here](https://docs.amplify.aws/lib/project-setup/prereq/q/platform/ios) & [here](https://docs.amplify.aws/lib/auth/getting-started/q/platform/ios).
2. Configure AWS Cognito with "Sign In with Apple" as explained [here](https://docs.amplify.aws/lib/auth/social_signin_web_ui/q/platform/ios#setup-your-auth-provider).
3. Run on a real device.
4. Tap on 'Sign In With Apple' button. which calls
```SWIFT
Amplify.Auth.signInWithWebUI(for: .apple, presentationAnchor: window,listener: completionHandler)
```
5. See the native sign-in bottom sheet.

**Expected behavior**
The Bottom Sheet should show the app/target name from the project settings, as well as the app icon from the assets folder exactly as `ASAuthorizationController`

**Screenshots**
Please check the screenshot below:
![image](https://user-images.githubusercontent.com/42779190/110497372-0db65100-80ff-11eb-9c3d-7077c9a67840.png)

**Environment**
 - Amplify (1.6.1):
    - Amplify/Default (= 1.6.1)
  - Amplify/Default (1.6.1)
  - AmplifyPlugins/AWSCognitoAuthPlugin (1.6.1):
    - AWSAuthCore (~> 2.23.0)
    - AWSCognitoIdentityProvider (~> 2.23.0)
    - AWSCognitoIdentityProviderASF (~> 2.23.0)
    - AWSCore (~> 2.23.0)
    - AWSMobileClient (~> 2.23.0)
    - AWSPluginsCore (= 1.6.1)
 - Dependency Manager: Cocoapods
 - Swift Version: 5
 - CLI Version: 4.41.2

**Device Information (please complete the following information):**
 - Device: iPhone 11 (iPhone12,3)
 - iOS Version: iOS 14.3

**Comment by lawmicha** [comment]
Hi @ameedsayeh thanks for opening the issue. We've seen this before and believe it was an Apple issue that was fixed. We'll try to reproduce this to see if there was a regression.

**Comment by palpatim** [comment]
I'm able to reproduce this in a sample app on simulators running both iOS 13 and iOS 14. However, I cannot reproduce this on my test devices (both of which are running iOS 14.4).

Can folks confirm that you are seeing this on physical devices? I see the original report is for 

> * Device: iPhone 11 (iPhone12,3)
> * iOS Version: iOS 14.3

**Comment by palpatim** [comment]
I'm also not seeing the behavior even on simulators if I call `Amplify.signInWithWebUI()` without passing a `for: provider` argument. This flavor of the API launches the HostedUI "landing page", where a user can select amongst all of the social providers you have set up as Identity Providers for your Cognito User Pool and enabled for your App Client. However, I note that in this case, the behavior is slightly different: On simulators, the messages is "Create an account for us-west-2.amazoncognito.com using your Apple ID...", while on the physical device, the message is "Create an account for "Test services ID" using your Apple ID...", where "Test Services ID" is the "Description" of the services ID I created for my Sign In With Apple configuration.

![image](https://user-images.githubusercontent.com/227982/110714028-b6c08280-81b7-11eb-8d56-475c1268f0e0.png)

**Comment by ameedsayeh** [comment]
> I'm able to reproduce this in a sample app on simulators running both iOS 13 and iOS 14. However, I cannot reproduce this on my test devices (both of which are running iOS 14.4).
> 
> Can folks confirm that you are seeing this on physical devices? I see the original report is for
> 
> > * Device: iPhone 11 (iPhone12,3)
> > * iOS Version: iOS 14.3
> 
> But I'd like explicit confirmation that this was seen on a physical device. If others are seeing this on physical devices, please leave a reply on this thread with the device model and iOS version.
> 
> @ameedsayeh
> 
> > not sure if this is related, but after having the login flow working yesterday. today it is not :(
> > apple page is showing with invalid_client error
> 
> Could you please open a new issue with the Bug Report template so we can get full details including repro steps? I'm not convinced it's the same issue, but we'll want to investigate it.

Seems like this part of the issue is not related to Amplify as it is reproducible without Amplify, the problem should be from Apple side.

I created an issue on Apple Developer forums to discuss that: 
https://developer.apple.com/forums/thread/675742

**Comment by ameedsayeh** [comment]
I can't also reproduce the main issue mentioned in the topic above as it is blocked/pending a reply to invalid_client issue from Apple side.

**Comment by ameedsayeh** [comment]
> I'm also not seeing the behavior even on simulators if I call `Amplify.signInWithWebUI()` without passing a `for: provider` argument. This flavor of the API launches the HostedUI "landing page", where a user can select amongst all of the social providers you have set up as Identity Providers for your Cognito User Pool and enabled for your App Client. However, I note that in this case, the behavior is slightly different: On simulators, the messages is "Create an account for us-west-2.amazoncognito.com using your Apple ID...", while on the physical device, the message is "Create an account for "Test services ID" using your Apple ID...", where "Test Services ID" is the "Description" of the services ID I created for my Sign In With Apple configuration.
> 
> ![image](https://user-images.githubusercontent.com/227982/110714028-b6c08280-81b7-11eb-8d56-475c1268f0e0.png)

I have a question if you don't mind.
the service ID in the picture, did you create it recently or it was created since long time?

**Comment by ameedsayeh** [comment]
**UPDATE**: Apple fixed the invalid_client issue and now I confirm that the flow is working and I can see the app name in the SIWA sheet, but I still can't see the icon/logo, do I need to do some extra configurations for the icon?

**Comment by palpatim** [comment]
@ameedsayeh Thanks for confirming. I haven't confirmed the "null" app name issue on my side--simulators still show a "null" app name, and I've never reproduced that issue on test devices. Likewise, I never encountered the "invalid_client" issue you report. I'm afraid I don't have insight into the app icon behavior either, but I wouldn't expect it would take a deployment to AppStore to have that icon show up.

We've double checked with the Cognito team, and they report that they haven't made any recent configuration or code changes that would impact this behavior. We're continuing to investigate and will update this issue as we get more information.

**Comment by jgale** [comment]
@ameedsayeh I am in the same boat. I can see the App name (as taken from the Services ID) but no icon (both on Simulator and physical device). We do have the app on App Store Connect (for TestFlight) but it hasn't been submitted to the public App Store. I also do not see the icon in the Hosted UI.

When I try using Sign in with Apple without Amplify/Cognito (i.e. via SwiftUI `SignInWithAppleButton`) it does properly show the icon.

If anyone has guidance on how to get the icon to show up, it would be greatly appreciated.

**Comment by ameedsayeh** [comment]
@palpatim,
On real devices: I can see the app name, but not the logo/icon.
On simulators: I can't see both.
Thanks!

**Comment by ameedsayeh** [comment]
@jgale, we'll release it to the store very soon, I will update if the icon issue was resolved or not.

**Comment by palpatim** [comment]
> On real devices: I can see the app name, but not the logo/icon.
> 
> On simulators: I can't see both.

Thanks for confirming. We'll update as soon as we have any information, but so far, we don't have any leads on this.

**Comment by ameedsayeh** [comment]
@jgale , we release the app to the store but unfortunately, the icon issue is still there :( 
Seems like if you have the 'SIWA capability' enabled to a group of apps (not a single app) the icon won't appear.

**Comment by palpatim** [comment]
> Seems like if you have the 'SIWA capability' enabled to a group of apps (not a single app) the icon won't appear.

Thanks for that clarification, @ameedsayeh.

**Comment by palpatim** [comment]
We don't believe this is an issue with either Cognito or the SDK. I've opened a feedback to Apple and will update this issue when I have any updated information.

**Comment by palpatim** [comment]
We have reached out to Apple for any information they can provide on this. We'll reopen if we get any updates.

**Comment by jgale** [comment]
> Seems like if you have the 'SIWA capability' enabled to a group of apps (not a single app) the icon won't appear.

@ameedsayeh, just to clarify what this means, under:
* Certificates, Identifiers & Profiles > App IDs > your-app-id > Sign in with Apple > Edit

do you have multiple apps with "Group with an existing primary App ID"? 

I have 3 different apps, and they are all marked as "Enable as a primary App ID". However, under:
* Certificates, Identifiers & Profiles > Service IDs 

I also have 3 separate Service IDs, one for each app. When I go into one of those, then click "Configure" on Sign in with Apple it shows all 3 of my App IDs under the Primary App ID drop down, and I select the primary one.

I don't see the SIWA icon on any of them.

**Comment by geekabhinavp** [comment]
# Fix for "Sign in with Apple" in AWS Cognito/Amplify  
(null app name, no icon, or **Sign up not complete**)

I ran into a nasty issue where **Sign in with Apple** worked in staging but failed in production.  
The Apple consent sheet showed a `null` app name with no icon, and Cognito gave me **“Sign up not complete.”**

After debugging, here’s the root cause and the full fix.

---

## The root cause
- I mistakenly used my **Bundle ID** as the `Client ID` in Cognito’s Apple IdP setup.  
- Cognito actually requires the **Apple Service ID** (the identifier you create in Apple Developer under *Identifiers → Service IDs*).  
- Without the proper Service ID + domain/return URLs, Apple doesn’t send all attributes (like `email`), and Cognito fails to complete signup.

---

## Step 1: Apple Developer setup

1. **Create a Service ID**  
   Example: `your.app.serviceid`

2. **Configure Web Authentication**  
   - **Domains and Subdomains**:
     ```
     <your-cognito-domain>.auth.<region>.amazoncognito.com
     ```
   - **Return URLs**:
     ```
     https://<your-cognito-domain>.auth.<region>.amazoncognito.com/oauth2/idpresponse
     ```
   - Verify the domain.

3. **Associate Primary App**  
   - Link the Service ID with your **Primary App (Bundle ID)** so Apple knows which app name/icon to display.

4. **App Store Connect**  
   - Make sure the Primary App exists with an uploaded app icon.  
   - Otherwise, Apple shows a blank icon in the consent screen.

---

## Step 2: Cognito setup

1. **Identity provider → Apple**
   - **Client ID** = your **Service ID** ✅  
     *(not the Bundle ID)*  
   - **Team ID**, **Key ID**, and `.p8` key as usual.
   - **Scopes**: `email`, `name`.

2. **Attribute mapping (Apple → Cognito)**
   - `email` (IdP) → `email` (Cognito)  
   - `email_verified` (IdP) → `email_verified` (Cognito)

3. **App client settings (Hosted UI)**
   - Callback URL:
     ```
     https://<your-cognito-domain>.auth.<region>.amazoncognito.com/oauth2/idpresponse
     ```
   - Sign-out URL: your app landing page (e.g. `https://your-app.com/`).  
   - Allowed OAuth flows: **Authorization code**.  
   - Allowed OAuth scopes: `openid`, `email`, `profile`.

---

## Step 3: Amplify frontend config

In `aws-exports.js` (or Amplify configure block):

```js
"oauth": {
  "domain": "<your-cognito-domain>.auth.<region>.amazoncognito.com",
  "scope": ["openid", "email", "profile"],
  "redirectSignIn": "https://your-app.com/",
  "redirectSignOut": "https://your-app.com/",
  "responseType": "code"
}
````

---

## Step 4: Reset stuck accounts

* Go to [[appleid.apple.com](https://appleid.apple.com/)](https://appleid.apple.com) → *Sign in with Apple* → remove your app so Apple will re-prompt for consent.
* Delete any half-created **UNCONFIRMED** users in Cognito.
* Try again in a private browser window.

---

## ✅ Result

* Apple consent sheet shows the correct **app name and icon**.
* Cognito successfully creates the user with `email` and marks them **CONFIRMED**.
* The **“Sign up not complete”** error is gone.

---

### TL;DR

Make sure Cognito’s `Client ID` = **Apple Service ID**, not your Bundle ID.
Register your Cognito domain + `/oauth2/idpresponse` as the return URL in Apple.
Map `email` and `email_verified`. Reset test accounts to force Apple to resend attributes.

```
